### PR TITLE
Exclude test recording files

### PR DESCRIPTION
**Pull request by sima-zhu**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
Resolve https://github.com/Azure/azure-sdk-tools/issues/3126
The Pr is to speed up the generate-job-matrix step by minimizing the checkout files.
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1647572&view=results
If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by scbedd — APPROVED** [review]
Nice! Makes sense to me!

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -99,6 +99,8 @@ stages:
         jobs:
         - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
           parameters:
+            SparseCheckoutPaths:
+              - "sdk/${{ parameters.ServiceDirectory }}/**/*.json"
```
Comment (benbp) [inline eng/pipelines/templates/stages/archetype-sdk-tests.yml:103]: Might be worth a comment explaining that this implicitly excludes the yaml based recording files?

**Comment by sima-zhu** [comment]
@scbedd @benbp The PR is not able to override and checkin. Could you help check this in?

**Pull request merged**